dask/array/core.py

```diff
@@ -3470,15 +3470,34 @@ def _vindex(x, *indexes):
     array([ 0,  9, 48,  7])
     """
     indexes = replace_ellipsis(x.ndim, indexes)
-    partial_slices = {i: ind for i, ind in enumerate(indexes)
-                      if isinstance(ind, slice) and ind != slice(None)}
+
+    num_indexes = []
+    partial_slices = []
+    reduced_indexes = []
+    for i, ind in enumerate(indexes):
+        if isinstance(ind, Number):
```

Comment:
I would suggest checking for 0-dimensional arrays here, too, e.g., `if isinstance(ind, Number) or getattr(ind, 'ndim', None) == 0`

Reply:
Yeah, was thinking about that too. Will update.

Reply:
This might be tricky though if the object being checked is lazy. We can check for Dask Arrays pretty easily. Not sure if we should be checking for other things.